src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/RuntimeEnvironment.cs

```diff
@@ -22,8 +22,8 @@ public static bool FromGlobalAccessCache(System.Reflection.Assembly a)
             return false;
         }
         public static string GetRuntimeDirectory()
-        {
-            return Path.GetDirectoryName(typeof(object).Assembly.Location) + Path.DirectorySeparatorChar;
+        { 
```

Comment:
I think you should still call Assembly.Location. If it returns empty string or non-absolute path, either throw or fallback to AppDomain.BaseDirectory.